lib/db_utils.go

```diff
@@ -1482,159 +1533,153 @@ func DBGetLimitedMessageForMessagingKeys(handle *badger.DB, messagingKeys []*Mes
 			"problem enumerating messaging key entries for prefix (%v)", prefix)
 	}
 
-	// Decode found messaging key entries.
-	messagingKeyEntries := []*MessagingGroupEntry{}
+	// Decode found access key entries.
+	var accessKeyEntries []*AccessGroupEntry
 	for _, valBytes := range valuesFound {
-		messagingKeyEntry := &MessagingGroupEntry{}
+		accessKeyEntry := &AccessGroupEntry{}
 		rr := bytes.NewReader(valBytes)
-		if exists, err := DecodeFromBytes(messagingKeyEntry, rr); !exists || err != nil {
-			return nil, errors.Wrapf(err, "DBGetMessagingGroupEntriesForOwnerWithTxn: "+
+		if exists, err := DecodeFromBytes(accessKeyEntry, rr); !exists || err != nil {
+			return nil, errors.Wrapf(err, "DBGetAccessGroupEntriesForOwnerWithTxn: "+
 				"problem decoding messaging key entry for public key (%v)", ownerPublicKey)
 		}
 
-		messagingKeyEntries = append(messagingKeyEntries, messagingKeyEntry)
+		accessKeyEntries = append(accessKeyEntries, accessKeyEntry)
 	}
 
-	return messagingKeyEntries, nil
+	return accessKeyEntries, nil
 }
 
-func DBGetAllUserGroupEntiresWithTxn(txn *badger.Txn, ownerPublicKey []byte) ([]*MessagingGroupEntry, error) {
-	// This function fetches all MessagingGroupEntries for the user from the DB. This includes the
+func DBGetAllAccessGroupEntriesForMemberWithTxn(txn *badger.Txn, snap *Snapshot, memberPublicKey []byte) ([]*AccessGroupEntry, error) {
+	// This function fetches all AccessGroupEntries for the user from the DB. This includes the
 	// base entry, the owner group entries, and the member group entries.
 
 	// We will keep track of all group entries in this array.
-	var userGroupEntries []*MessagingGroupEntry
+	var userGroupEntries []*AccessGroupEntry
 
-	// First add the base messaging key.
-	userGroupEntries = append(userGroupEntries, &MessagingGroupEntry{
-		GroupOwnerPublicKey:   NewPublicKey(ownerPublicKey),
-		MessagingPublicKey:    NewPublicKey(ownerPublicKey),
-		MessagingGroupKeyName: BaseGroupKeyName(),
+	// First add the base access key.
+	userGroupEntries = append(userGroupEntries, &AccessGroupEntry{
+		GroupOwnerPublicKey: NewPublicKey(memberPublicKey),
+		AccessPublicKey:     NewPublicKey(memberPublicKey),
+		AccessGroupKeyName:  BaseGroupKeyName(),
 	})
 
-	// Now add all the groups where this user is the owner
-	ownerGroupEntries, err := DBGetMessagingGroupEntriesForOwnerWithTxn(txn, NewPublicKey(ownerPublicKey))
-	if err != nil {
-		return nil, errors.Wrapf(err, "DBGetAllUserGroupEntiresWithTxn: problem getting messaging entries")
-	}
-	userGroupEntries = append(userGroupEntries, ownerGroupEntries...)
-
 	// And add the groups where the user is a member
-	memberGroupEntries, err := DBGetAllMessagingGroupEntriesForMemberWithTxn(txn, NewPublicKey(ownerPublicKey))
+	// Never use the deprecated function for backwards compatibility here because of no access to blockHeight
+	accessGroupEntries, err := DBGetAllEntriesForPublicKeyFromMembershipIndexWithTxn(txn, snap, NewPublicKey(memberPublicKey))
```

Comment:
This won't work if we're before the fork height, correct? If so, we'll prob need branching logic to call a version of this function with the pre-fork logic unfortunately.

Reply:
Yes, sorry it's not clear in this function but we do have branching logic in place. I'll put in some comments here. Basically this function is being called only if we are already after the fork height. If we are before the fork height, there is a separate function being called that begins with "DEPRECATED".